tests/NationalInstruments.Analyzers.UnitTests/ChainOfMethodsWithLambdasAnalyzerTests.cs

```diff
@@ -0,0 +1,648 @@
+    }
+}
+", GetNI1017Rule());
+
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_WellSplitLambdasExtensionMethods_NoDiagnostic()
+        {
+            var test = new AutoTestFile(
+@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(ISoftwareContent softwareContent)
+        {
+            softwareContent.Foo(x => true)
+                .Baz(x => true);
+        }
+
+    }
+    public interface ISoftwareContent
+    {
+    }
+
+    public static class ISoftwareContentExtensions
+    {
+        public static ISoftwareContent Foo(
+            this ISoftwareContent softwareContent,
+            Predicate<string> predicate) => softwareContent;
+
+        public static ISoftwareContent Baz(
+            this ISoftwareContent softwareContent,
+            Predicate<string> predicate) => softwareContent;
+    }
+}
+");
+
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_PoorlySplitLambdasDelegateInvocations_EmitsDiagnostic()
+        {
+            var test = new AutoTestFile(
+@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    public delegate ISoftwareContent Creator(Func<ISoftwareContent, string> func);
+
+    class Test
+    {
+
+        void Foo(ISoftwareContent softwareContent)
+        {
+            <|softwareContent.Foo(soft => ""first lambda"").Foo(soft => ""second lambda"")|>;
+        }
+
+    }
+    public interface ISoftwareContent
+    {
+        Creator Foo { get; }
+    }
+}
+", GetNI1017Rule());
+
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_WellSplitLambdasDelegateInvocations_NoDiagnostic()
+        {
+            var test = new AutoTestFile(
+@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    public delegate ISoftwareContent Creator(Func<ISoftwareContent, string> func);
+
+    class Test
+    {
+
+        void Foo(ISoftwareContent softwareContent)
+        {
+            softwareContent.Foo(soft => ""first lambda"")
+                .Foo(soft => ""second lambda"");
+        }
+
+    }
+    public interface ISoftwareContent
+    {
+        Creator Foo { get; }
+    }
+}
+");
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_WhitespaceBetweenMultipleLambdasInMethodInvocations_EmitsDiagnostic()
+        {
+            var test = new AutoTestFile(
+@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(ISoftwareContent softwareContent)
+        {
+            <|softwareContent         .Children(x => true)          .Any(s => s.AliasName == ""T"")|>;
```

Comment:
How does this yield a diagnostic if you're just checking for leading trivia?

Reply:
Such whitespaces become [`TrailingTrivia`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.syntaxtoken.trailingtrivia?view=roslyn-dotnet) of the previous `SyntaxNode`.

![image](https://user-images.githubusercontent.com/71984539/97040648-ddc14380-158b-11eb-94b5-e9316730103d.png)